fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -1169,5 +1178,33 @@ private void handleExportStmt() throws Exception {
     private List<PrimitiveType> exprToType(List<Expr> exprs) {
         return exprs.stream().map(e -> e.getType().getPrimitiveType()).collect(Collectors.toList());
     }
+
+    private void matchSql(QueryStmt queryStmt) throws AnalysisException {
+        Map<String, List<SqlBlockRule>> userToSqlBlockRuleMap = Catalog.getCurrentCatalog().getSqlBlocklistMgr().getUserToSqlBlockRuleMap();
```

Comment:
If you get `userToSqlBlockRuleMap` out of the `SqlBlocklistMgr`, then no lock can protect it.
So all check should be done inside the `SqlBlocklistMgr`